js/app/warehouse.js

```diff
@@ -32,10 +33,29 @@ var Warehouse = new Knowledge.Class({
                     return current_dir[file_name][module_path];
                 }
                 return current_dir[file_name];
-            }, imports.app.modules);
+            }, importer);
         } catch (error if (error.message.startsWith('No JS module'))) {
```

Comment:
optional: While you're touching this code, consider getting rid of the `catch (error if condition)` since I recently learned the `if` clause is nonstandard and will be removed from the JS engine